commit cb3fa6852d8fb6f5dfaab6a416ff3b87d7b72c59 (HEAD -> refs/heads/file_walking)
Author: Вадим Кузнецов <rics@ideco.ru>
Date:   Thu Feb 22 10:29:13 2018 +0500

    Sorting files names alphabetically

diff --git a/src/main.c b/src/main.c
index 2671a21..0172ee4 100644
--- a/src/main.c
+++ b/src/main.c
@@ -239,6 +239,8 @@ int main(int argc, char** argv)
     }
   }
 
+  imv_navigator_sort(&nav);
+
   /* if we weren't given any paths we have nothing to view. exit */
   if(!imv_navigator_selection(&nav)) {
     fprintf(stderr, "No input files. Exiting.\n");
diff --git a/src/navigator.c b/src/navigator.c
index 97772cd..8ab492f 100644
--- a/src/navigator.c
+++ b/src/navigator.c
@@ -22,6 +22,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 #include <dirent.h>
 #include <stdlib.h>
 #include <string.h>
+#include <strings.h>
 #include <stdio.h>
 
 void imv_navigator_init(struct imv_navigator *nav)
@@ -48,6 +49,30 @@ void imv_navigator_destroy(struct imv_navigator *nav)
   memset(nav, 0, sizeof(struct imv_navigator));
 }
 
+static int check_ext(const char *path)
+{
+  static const char *const valid_exts[] = {
+    ".gif",
+    ".ico",
+    ".jpeg",
+    ".jpg",
+    ".png",
+    ".tif",
+    ".tiff",
+  };
+
+  char *const ext = strrchr(path, '.');
+
+  if (ext == NULL)
+    return 1;
+
+  for (size_t i = 0; i < (sizeof(valid_exts) / sizeof(char*)); i++)
+    if (strcasecmp(ext, valid_exts[i]) == 0)
+      return 0;
+
+  return 1;
+}
+
 static int add_item(struct imv_navigator *nav, const char *path,
                      time_t mtime)
 {
@@ -95,6 +120,8 @@ int imv_navigator_add(struct imv_navigator *nav, const char *path,
             return 1;
           }
         } else {
+          if (check_ext(path_buf) != 0)
+            continue;
           stat(path_buf, &path_info);
           if(add_item(nav, path_buf, path_info.st_mtim.tv_sec) != 0) {
             return 1;
@@ -104,6 +131,8 @@ int imv_navigator_add(struct imv_navigator *nav, const char *path,
       closedir(d);
     }
   } else {
+    if (check_ext(path) != 0)
+        return 0;
     return add_item(nav, path, path_info.st_mtim.tv_sec); 
   }
 
@@ -245,5 +274,15 @@ int imv_navigator_wrapped(struct imv_navigator *nav)
   return nav->wrapped;
 }
 
+static int cmp_names(const void *first, const void *second)
+{
+  return strcasecmp(*(char *const *) first, *(char *const *) second);
+}
+
+void imv_navigator_sort(struct imv_navigator *nav)
+{
+  if (nav->num_paths > 0)
+    qsort(&(nav->paths[0]), nav->num_paths, sizeof(char*), cmp_names);
+}
 
 /* vim:set ts=2 sts=2 sw=2 et: */
diff --git a/src/navigator.h b/src/navigator.h
index ac3cc59..44cc609 100644
--- a/src/navigator.h
+++ b/src/navigator.h
@@ -70,6 +70,9 @@ int imv_navigator_poll_changed(struct imv_navigator *nav, const int nopoll);
 /* Check whether navigator wrapped around paths list */
 int imv_navigator_wrapped(struct imv_navigator *nav);
 
+/* Sort all added files alphabetically. */
+void imv_navigator_sort(struct imv_navigator *nav);
+
 #endif
 
 
